ui/src/widgets/ui-chart/UIChart.vue

```diff
@@ -24,33 +24,54 @@ export default {
             /** @type {Chart} */
             chart: null,
             hasData: false,
+            histogram: [], // populate later for bins per series
             chartUpdateDebounceTimeout: null
         }
     },
     computed: {
         ...mapState('data', ['messages']),
+        chartType () {
+            // ChartJS doesn't support Histograms, so we render it as a bar chart,
+            // but maintain the ref to our own config
+            if (this.props.chartType === 'histogram') {
+                return 'bar'
+            }
+            return this.props.chartType
+        },
+        xAxisType () {
+            // we use this to decide on some custom calculations of the categories in Histograms,
+            // but ChartJS needs the 'category; type for the axis still
+            if (this.props.xAxisType === 'bins') {
+                return 'category'
+            }
+            return this.props.xAxisType
+        },
         radialChart () {
             // radial charts have no placeholder in ChartJS - we need to add one
             return this.props.xAxisType === 'radial'
         }
     },
     watch: {
+        chartType: function (value) {
+            this.chart.config.type = value
+            this.update()
+        },
         'props.label': function (value) {
             this.chart.options.plugins.title.text = value
-            this.chartUpdate(false)
+            this.update()
```

Comment:
```suggestion
            this.update(false)
```
As this is a prop update, should it not also use `false` (i.e. debounce multiple quick successive prop changes)

Reply:
As discussed with @Steve-Mcl applying the suggestions